administration/src/bp-modules/applications/NoteDialogController.tsx

```diff
@@ -0,0 +1,50 @@
+import React, { ReactElement } from 'react'
+
+import getMessageFromApolloError from '../../errors/getMessageFromApolloError'
+import { useGetApplicationByIdQuery, useUpdateApplicationNoteMutation } from '../../generated/graphql'
+import getQueryResult from '../../mui-modules/util/getQueryResult'
+import { useAppToaster } from '../AppToaster'
+import TextAreaDialog from '../components/TextAreaDialog'
+import { Application } from './ApplicationsOverview'
+
+type NoteDialogControllerProps = {
+  application: Application
+  onOpenNoteDialog: (value: boolean) => void
+  isOpen: boolean
+}
+
+const NoteDialogController = ({ application, onOpenNoteDialog, isOpen }: NoteDialogControllerProps): ReactElement => {
+  const appToaster = useAppToaster()
+  const [updateApplicationNote, { loading }] = useUpdateApplicationNoteMutation({
+    onError: error => {
+      const { title } = getMessageFromApolloError(error)
+      appToaster?.show({ intent: 'danger', message: title })
+    },
+    onCompleted: () => appToaster?.show({ intent: 'success', message: 'Notiz erfolgreich geändert.', timeout: 2000 }),
+  })
+  const applicationQuery = useGetApplicationByIdQuery({
+    variables: { applicationId: application.id },
+  })
+  const applicationQueryHandler = getQueryResult(applicationQuery)
+  if (!applicationQueryHandler.successful) return applicationQueryHandler.component
```

Comment:
Isn't the note already on the application prop so we don't actually need to run another query?

Reply:
yes but if its gonna be updated by the user, we would have to update all applications to keep it in sync. I thinks this should be avoided

Reply:
But probably i should block the fetching somehow until its called by the `onSave` method. 
Have to check how i can do this to avoid a first unneeded call or do you have a better solution @michael-markl ?

Reply:
ok i realized there are lazy query hooks that seem to be suitable